src/conversion/model-to-view-converters.js

```diff
@@ -334,7 +338,11 @@ export function wrap( elementCreator ) {
 
 		// Then wrap with the new wrapper.
 		if ( data.attributeNewValue !== null ) {
-			viewWriter.wrap( viewRange, newViewElement );
+			if ( data.item instanceof ModelSelection ) {
+				viewWriter.wrap( conversionApi.viewSelection.getFirstRange(), newViewElement, conversionApi.viewSelection );
+			} else {
+				viewWriter.wrap( viewRange, newViewElement );
+			}
```

Comment:
You remember that we agreed that this condition should be earlier, don't you?

Reply:
Now I do.